appcelerator/titanium_mobile
# fix: convertPointToView() to use "ti.ui.defaultunit"

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27807

**Summary:**
- Fixes bug where `convertPointToView()` always returns pixels on Android.
- Fixes bug where `convertPointToView()` always returns dips on iOS.
- This method now returns the units set by "tiapp.xml" file's ti.ui.defaultunit". _(Dips by default.)_
- Fixed `convertPointToView()` to accept x/y point string values with post-fixed units.

**Test:**
1. Build and run the below code on Android.
2. Attempt to drag one of the squares on screen.
3. Verify square follows your finger. _(Square may wobble on Android due to a separate issue.)_
4. Build for iOS and repeat steps 2 - 3.
5. In the "tiapp.xml", set the following property to pixels.
`<property name="ti.ui.defaultunit" type="string">px</property>`
6. Build for Android and repeat steps 2 -3.
7. Build for iOS and repeat steps 2 -3.

app.js
```javascript
function createDragSquare(top, left, color) {
	var view = Ti.UI.createView({
		touchEnabled: true,
		backgroundColor: color,
		center: { x: left + 50, y: top + 50 },
		width: 100,
		height: 100,
	});
	view.addEventListener("touchmove", function(event) {
		view.center = view.convertPointToView({ x: event.x, y: event.y }, window);
		event.cancelBubble = true;
	});
	return view;
}

var window = Ti.UI.createWindow({ backgroundColor: "white" });
window.add(createDragSquare(40, 40, "red"));
window.add(createDragSquare(120, 120, "green"));
window.add(createDragSquare(200, 200, "blue"));
window.open();
```
